walker.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""A Random Walk Simulation For Multiple Walkers"""
+
+# Python code for 2D random walk.
+# Source: https://www.geeksforgeeks.org/random-walk-implementation-python/
+import sys
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def create_walkers(number_of_steps, number_of_walkers):
+    """
+    Creates a list of numpy arrays which each hold the coordinates of a walker. The coordinates
+    are randomly chosen in a range from 0 to the number of steps
+    :param number_of_steps: the number of steps the walker will make. This defines the
+    dimension of the array which represents the walker
+    :param number_of_walkers: the number of walkers defines the number of arrays, which
+    will be created
+    :return: a list, the 'scene' which contains all walkers, so all arrays containing coordinates
+    """
+    scene = []
+    for _ in range(number_of_walkers):
+        # create the array of a walker and add it to the scene-list
+        walker = np.random.randint(low=0, high=number_of_steps,
+                                    size=(number_of_steps, 2))
+        scene.append(walker)
+
+    return scene
+
+  
+def calculate_the_path(walker, number_of_steps):
+    """
+    Calculates the path the walker walks
+    :param walker: The array representing the walker, holding random coordinates
+    :param number_of_steps: the number of steps the walker takes
+    :return: the updated array representing the walker, holding all coordinates the walker walks to
+    """
+    # split coordinates
+    x_coord = walker[:, 0]
+    y_coord = walker[:, 1]
+
+    # calculate new coordinates for each step
+    for stepnumber in range(1, number_of_steps):
```

Comment:
Your walker seems to have an interesting behavior of either walking south _or_ north during execution without direction changes, while randomly switching between going left and right. I was not able to find the reason without debugging but it might be the case that your code never reaches either the walk north or the walk south case.

Reply:
hm, strange... will have a look into it if I find something

Reply:
yes, found the bug and fixed it! Thank you